src/Common/Config/ConfigProcessor.cpp

```diff
@@ -693,6 +693,8 @@ void ConfigProcessor::savePreprocessedConfig(const LoadedConfig & loaded_config,
 void ConfigProcessor::setConfigPath(const std::string & config_path)
 {
     main_config_path = config_path;
+    if (!main_config_path.ends_with('/'))
+        main_config_path += '/';
```

Comment:
The bug was introduced in [that commit](https://github.com/ClickHouse/ClickHouse/commit/8bac10d24f3c76563a478f2c8b3492975dd65034#diff-7a37a2cdf7bff83ad14829055d0eab3abee67c1c671716d9116ed7c53cce68a3R473).

Explanation: `Poco::Path(config_path).makeParent().toString()` returns the parent path with a slash at the end, and `fs::path(config_path).parent_path()` returns the parent path without a slash.